contracts/0.8.9/StakingRouter.sol

```diff
@@ -0,0 +1,546 @@
+// SPDX-FileCopyrightText: 2022 Lido <info@lido.fi>
+
+// SPDX-License-Identifier: GPL-3.0
+
+/* See contracts/COMPILERS.md */
+pragma solidity 0.8.9;
+
+import {AccessControlEnumerable} from "@openzeppelin/contracts-v4.4/access/AccessControlEnumerable.sol";
+
+import {ILido} from "./interfaces/ILido.sol";
+import {IStakingRouter} from "./interfaces/IStakingRouter.sol";
+import {IStakingModule} from "./interfaces/IStakingModule.sol";
+import {IDepositContract} from "./interfaces/IDepositContract.sol";
+
+import {Math} from "./lib/Math.sol";
+import {BatchedSigningKeys} from "./lib/BatchedSigningKeys.sol";
+import {UnstructuredStorage} from "./lib/UnstructuredStorage.sol";
+import {MinFirstAllocationStrategy} from "./lib/MinFirstAllocationStrategy.sol";
+
+import {BeaconChainDepositor} from "./BeaconChainDepositor.sol";
+
+contract StakingRouter is IStakingRouter, AccessControlEnumerable, BeaconChainDepositor {
```

Comment:
add revert for recieve function

Reply:
Add something like `getStakingModuleMaxDepositableKeys` to allow tooling know how much keys is depositable.

This method should call `Lido.getAvailableToStake()`

Reply:
added `getStakingModuleMaxDepositableKeys(uint24 _stakingModuleId)`  to `Lido` contract since the staking router knows nothing about Lido